### PR TITLE
Code review for Bortoletti and Barca

### PR DESCRIPTION
**Pull request by albertoursino**
Hi guys I'm Alberto Ursino.
I commited 3 files to review: 1 MainActivity and 2 simple classes.

### CODE REVIEW

**Review by giobbarca — COMMENTED** [review]
Personal opinion: don't use a timeout for permission request but "sleep" the application until he doesn't give the permission (or user can't use the app until he doesn't grant permissions). Avoid closing app if timeout expired.

Good overall formattation, good coding, no hardwiring.
Some advices for MainActivity.java are listed below. Other two classes are fine.

MainActivity.java

- Row 128: is ok to use finish() and System.exit() or the application should be more "user-friendly" (asking again for permission or something else..)
- Row 178: throwed an exception but no JavaDoc spec (@throws) given

File: app/src/main/java/com/gruppo4/RingApplication/MainActivity.java [hunk 1]
```diff
@@ -8,16 +8,12 @@
...
 import android.widget.Button;
 import android.widget.EditText;
 import android.widget.ImageButton;
-import android.widget.Spinner;
 import android.widget.Toast;
-
 import androidx.annotation.NonNull;
 import androidx.appcompat.app.AppCompatActivity;
-
 import com.gruppo4.RingApplication.ringCommands.AppManager;
```
[the remark this answers is not in the document]
Comment (lettomobile) [reply app/src/main/java/com/gruppo4/RingApplication/MainActivity.java]: 1) I don't understand what do you mean with "don't use a timeout for permission request...".
I don't use any timeout.. 
2) I saw what you edited: I corrected what you said about "onClick". Do you think it's better now?
3) I grouped all imports! thanks.
4) Ok I can make my application more "user-friendly" but it's a demo so I can improve this thing in future...
5) I added the JavaDoc spec for the exception... thanks.

**Comment by bortoletti-giorgia** [comment]
Class RingCommand:
- Only a question: are specifications to have valid password written in another class?

Class PasswordManager:
- I suggest you to save the Context as a private variable so every time you don't have to ask for it as parameter. However I don't think if it is the best choice.

Class MainActivity:
- I saw the changes after the other code review and I have nothing to add.

In general good comments, no hardwiring, good formattation and good code.

Nice PREFERENCES_PASSWORD_KEY.

**Comment by lettomobile** [comment]
1) I don't understand your question, I ask you if you can clarify better...
2) Excellent observation on the context repeated many times, I corrected.

**Comment by lettomobile** [comment]
> In the class RingCommand at the line 22 it is written “valid password” for the parameter “password”. What do you mean with “valid”?
> It is only a curiosity to know if you check the validity in another class before invoking the method.

Good observation, I corrected and improved some PasswordManager instantiation

**Pull request merged**